### PR TITLE
agent: apply CPU constraints to containers

### LINKED ISSUE #203: missing support for CPU cgroups
agent should be able to apply CPU constraints to containers

### PR DESCRIPTION
**Pull request by devimc**
this patch reads the CPU constrains from the ctl channel and apply them

fixes #203

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

**Comment by devimc** [comment]
depends on https://github.com/containers/virtcontainers/pull/581
https://github.com/clearcontainers/packaging/pull/256
https://github.com/clearcontainers/proxy/pull/208

File: agent.go [hunk 1]
```diff
@@ -1355,6 +1355,8 @@ func newContainerCb(pod *pod, data []byte) error {
 				MemorySwappiness: nil,
 				AllowAllDevices:  nil,
 				AllowedDevices:   configs.DefaultAllowedDevices,
+				CpuQuota:         int64(payload.Constraints.CPUQuota),
```
Comment (jodh-intel) [inline agent.go]: I guess it's unlikely to be an issue, but this _could_ overflow, so could be written as:

```
if payload.Constraints.CPUQuota > uint64(math.MaxInt64) {
    return fmt.Errorf("CPU quota too big: %d", payload.Constraints.CPUQuota)
}

cpuQuota := int64(payload.Constraints.CPUQuota)
```

File: api/commands.go [hunk 2]
```diff
@@ -173,6 +173,15 @@ type SystemMountsInfo struct {
 	DevShmSize int `json:"devShmSize"`
 }
 
+// Constraints describes the constrains for a container
```
Comment (egernst) [inline api/commands.go]: s/constrains/constraints

Comment (devimc) [reply api/commands.go]: @egernst  fixed, thanks

**Review by sboeuf — APPROVED** [review]

**Comment by sboeuf** [comment]
LGTM

**Pull request merged**